### PR TITLE
Tests about support of HTML tags in translations keys with GetText

### PR DESCRIPTION
**Pull request by looorent**
Using this library, we have noticed translation keys including HTML tags were splited into multiple translation keys in .po files.

After writing a few tests about HTML tags and the PoCompiler, we identified the problem came from an old version of `gettext-parser` included in our project. So, if you want to include these test in your codebase, here are them.

Pay attention we've added a dirty fix in `tsconfig.json` to support the use of `gettext-parser` import with mocha: `"noImplicitAny": true`

### CODE REVIEW

**Comment by biesbjerg** [comment]
Thanks! I'm not crazy about changing `noImplicitAny` to `true`. Any way to avoid this, you think?

**Comment by looorent** [comment]
Actually, I believe `declaration.d.ts` should do the trick. I have no idea why `mocha -r` does not take it into account.

I think the best way to handle this is to augment `gettext-parser` with definition types. I've created a PR here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43325 

Once it is accepted, I will push a new version of this PR to support these types.

**Comment by looorent** [comment]
Now gettext-parser has a @types package (https://www.npmjs.com/package/@types/gettext-parser), here is an implementation without any `declaration.d.ts`.

File: src/compilers/po.compiler.ts [hunk 1]
```diff
@@ -34,23 +34,24 @@ export class PoCompiler implements CompilerInterface {
 			}
 		};
 
-		return gettext.po.compile(data);
+		return po.compile(data).toString("utf8");
```
Comment (biesbjerg) [inline src/compilers/po.compiler.ts:37]: Could you please change to use single quotes?

Comment (biesbjerg) [reply src/compilers/po.compiler.ts:37]: Note: I've added a single quote rule to tslint config in master

**Comment by looorent** [comment]
I've already removed `declaration.d.ts` in my previous commit, do you prefer to keep it empty?
The double quotes have been changed.

**Comment by biesbjerg** [comment]
Thanks!

**Comment by silk-bahamut** [comment]
Hello, I have a text with some HTML inside but it always ends up split in multiple keys.
I'm using the last version (7.0.0), extracting to json (but I tried with pot output too).

My text is looking something like that :
`<p translate>
   I'm so happy, <b>Woot</b>,
  I jump around
</p>`

Is there a limitation on which HTML tags are included or a configuration to do in order to active it  ?
The version of gettext-parser is 4.0.3 in my yarn.lock.

Could you please help me ?

**Pull request merged**